src/geo/transform.js

```diff
@@ -305,6 +325,32 @@ class Transform {
         this._cameraZoom = this._zoomFromMercatorZ(terrainElevation + height);
     }
 
+    sampleAverageElevation(): number {
+        if (!this._elevation) return 0;
+
+        const horizon = this.horizonLineFromTop();
+
+        let elevationSum = 0.0;
+        let weightSum = 0.0;
+        if (this._elevation) {
```

Comment:
This check is done earlier in the function, it can be removed.

Reply:
Sadly, no. Because there is a function call between the two (`horizonLineFromTop`), flow can't guarantee that `this._elevation` is still non-null.

Reply:
Ah ok I see, I tried the following and it seems to work fine:
```diff
diff --git a/src/geo/transform.js b/src/geo/transform.js
index 25f074f7a..f9f1968e0 100644
--- a/src/geo/transform.js
+++ b/src/geo/transform.js
@@ -319,6 +319,7 @@ class Transform {
 
     sampleAverageElevation(): number {
         if (!this._elevation) return 0;
+        const elevation: Elevation = this._elevation;
 
         const elevationSamplePoints = [
             [0.5, 0.2],
@@ -332,19 +333,17 @@ class Transform {
 
         let elevationSum = 0.0;
         let weightSum = 0.0;
-        if (this._elevation) {
-            for (let i = 0; i < elevationSamplePoints.length; i++) {
-                const pt = new Point(
-                    elevationSamplePoints[i][0] * this.width,
-                    horizon + elevationSamplePoints[i][1] * (this.height - horizon)
-                );
-                const hit = this._elevation.pointCoordinate(pt);
-                if (!hit) continue;
-
-                const weight = 1 / Math.hypot(hit[0] - this._camera.position[0], hit[1] - this._camera.position[1]);
-                elevationSum += hit[3] * weight;
-                weightSum += weight;
-            }
+        for (let i = 0; i < elevationSamplePoints.length; i++) {
+            const pt = new Point(
+                elevationSamplePoints[i][0] * this.width,
+                horizon + elevationSamplePoints[i][1] * (this.height - horizon)
+            );
+            const hit = elevation.pointCoordinate(pt);
+            if (!hit) continue;
+
+            const weight = 1 / Math.hypot(hit[0] - this._camera.position[0], hit[1] - this._camera.position[1]);
+            elevationSum += hit[3] * weight;
+            weightSum += weight;
         }
 
         if (weightSum === 0) return 0;

````

Reply:
Perfect. I figured wasn't worth trying since it'd mean an extra variable anyway, but this is better. Done. 👍